plugins/inputs/statsd/statsd.go

```diff
@@ -505,10 +507,17 @@ func (s *Statsd) parseName(bucket string) (string, string, map[string]string) {
 
 	var field string
 	name := bucketparts[0]
-	p, err := graphite.NewGraphiteParser(s.MetricSeparator, s.Templates, nil)
-	if err == nil {
-		p.DefaultTags = tags
-		name, tags, field, _ = p.ApplyTemplate(name)
+
+	p := s.graphiteParser
+	var err error
+
+	if p == nil {
+		p, err = graphite.NewGraphiteParser(s.MetricSeparator, s.Templates, nil)
+		if err == nil {
+			s.graphiteParser = p
+			p.DefaultTags = tags
+			name, tags, field, _ = p.ApplyTemplate(name)
```

Comment:
this part where the template gets applied should be outside of the `p == nil` if-statement